engines/group/app/controllers/group/api/v1/memberships_controller.rb

```diff
@@ -9,7 +9,8 @@ class MembershipsController < ApplicationController
         # GET /api/v1/memberships
         #
         def index
-          memberships = Membership.where(user_id: current_user.id)
+          group_ids = current_user.groups.map(&:id)
+          memberships = Membership.where(group_id: group_ids)
```

Comment:
nope, `current_user.groups.map(&:id)` does a `JOIN` loading all the columns from the table.

I propose:
```ruby
group_ids = Group::Membership.where(user_id: user.id).pluck(:group_id)
```
that does not need a `JOIN` and only loads the `id` column in memory.